custom_components/tahoma/climate_st.py

```diff
@@ -0,0 +1,242 @@
+
+    def __init__(self, device, controller, sensor):
+        """Init method."""
+        super().__init__(device, controller)
+        self._temp_sensor_entity_id = sensor
+        self._saved_target_temp = (
+            self.select_state(PRESET_TEMPERATURES[self.preset_mode])
+            if self.hvac_mode == HVAC_MODE_AUTO
+            else self.select_state(CORE_DEROGATED_TARGET_TEMPERATURE_STATE)
+        )
+        self._current_temperature = None
+
+    async def async_added_to_hass(self):
+        """Register temperature sensor after added to hass."""
+        await super().async_added_to_hass()
+        async_track_state_change(
+            self.hass, self._temp_sensor_entity_id, self._async_temp_sensor_changed
+        )
+
+        @callback
+        def _async_startup(event):
+            """Init on startup."""
+            if self._temp_sensor_entity_id is not None:
+                temp_sensor_state = self.hass.states.get(self._temp_sensor_entity_id)
+                if temp_sensor_state and temp_sensor_state.state != STATE_UNKNOWN:
+                    self.update_temp(temp_sensor_state)
+
+        self.hass.bus.async_listen_once(EVENT_HOMEASSISTANT_START, _async_startup)
+
+        self.schedule_update_ha_state(True)
+
+    async def _async_temp_sensor_changed(self, entity_id, old_state, new_state) -> None:
+        """Handle temperature changes."""
+        if new_state is None or old_state == new_state:
+            return
+        self.update_temp(new_state)
+        self.schedule_update_ha_state()
+
+    @callback
+    def update_temp(self, state):
+        """Update thermostat with latest state from sensor."""
+        if state is None or state.state == STATE_UNKNOWN:
+            return
+
+        try:
+            self._current_temperature = float(state.state)
+        except ValueError as ex:
+            _LOGGER.error("Unable to update from sensor: %s", ex)
+
+    @property
+    def temperature_unit(self) -> str:
+        """Return the unit of measurement used by the platform."""
+        return TEMP_CELSIUS
+
+    @property
+    def supported_features(self) -> int:
+        """Return the list of supported features."""
+        return SUPPORT_PRESET_MODE | SUPPORT_TARGET_TEMPERATURE
+
+    @property
+    def hvac_mode(self) -> str:
+        """Return hvac operation ie. heat, cool mode."""
+        return MAP_HVAC_MODES[self.select_state(ST_DEROGATION_TYPE_STATE)]
+
+    @property
+    def hvac_modes(self) -> List[str]:
+        """Return the list of available hvac operation modes."""
+        return [HVAC_MODE_AUTO, HVAC_MODE_HEAT]
+
+    @property
+    def preset_mode(self) -> Optional[str]:
+        """Return the current preset mode, e.g., home, away, temp."""
+        return MAP_PRESET_MODES[
+            self.select_state(ST_HEATING_MODE_STATE)
+            if self.hvac_mode == HVAC_MODE_AUTO
+            else self.select_state(ST_DEROGATION_HEATING_MODE_STATE)
+        ]
```

Comment:
Not really readable.

Reply:
is this better?
```
 @property
    def preset_mode(self) -> Optional[str]:
        """Return the current preset mode, e.g., home, away, temp."""
        st_preset = (
            self.select_state(ST_HEATING_MODE_STATE)
            if self.hvac_mode == HVAC_MODE_AUTO
            else self.select_state(ST_DEROGATION_HEATING_MODE_STATE)
        )
        return MAP_PRESET_MODES[st_preset]
```

Reply:
That's the "ternary" operator I don't like when on multiple line. A classic if block is more readable.
```python
@property
    def preset_mode(self) -> Optional[str]:
        """Return the current preset mode, e.g., home, away, temp."""
       state = self.select_state(ST_DEROGATION_HEATING_MODE_STATE)
       if self.hvac_mode == HVAC_MODE_AUTO
         state =  self.select_state(ST_HEATING_MODE_STATE)
       return MAP_PRESET_MODES[st_preset]
```

Reply:
let's agree to disagree on that ;-)
@iMicknl, can you help us choose between the solutions?